pkg/config/config.go

```diff
@@ -19,106 +19,68 @@ package config
 
 import (
 	"encoding/json"
+	"fmt"
 
+	"github.com/mariomac/pipes/pkg/graph"
+	"github.com/mariomac/pipes/pkg/graph/stage"
 	"github.com/netobserv/flowlogs-pipeline/pkg/api"
-	"github.com/sirupsen/logrus"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/decode"
+	"k8s.io/apimachinery/pkg/util/yaml"
 )
 
 type GenericMap map[string]interface{}
 
-var (
-	Opt        = Options{}
-	PipeLine   []Stage
-	Parameters []StageParam
-)
-
 type Options struct {
-	PipeLine   string
-	Parameters string
-	Health     Health
+	PipeLine PipelineDefinition
+	Health   Health
 }
 
 type Health struct {
 	Port string
 }
 
-type Stage struct {
-	Name    string
-	Follows string
-}
-
-type StageParam struct {
-	Name      string
-	Ingest    Ingest
-	Decode    Decode
-	Transform Transform
-	Extract   Extract
-	Encode    Encode
-	Write     Write
-}
-
-type Ingest struct {
-	Type      string
-	File      File
-	Collector api.IngestCollector
-	Kafka     api.IngestKafka
-	GRPC      api.IngestGRPCProto
+type PipelineDefinition struct {
+	graph.Connector
```

Comment:
https://github.com/mariomac/pipes/tree/main/examples/tutorial/02-highlevel-nodes#defining-the-graph-configuration